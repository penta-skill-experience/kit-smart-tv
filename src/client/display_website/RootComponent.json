{
  "ADMIN_PAGE_REFRESH_RATE": 20000
}
