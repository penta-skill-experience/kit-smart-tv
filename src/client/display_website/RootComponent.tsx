import * as React from "react";
import {SquareHolder} from "./SquareHolder";
export class RootComponent extends React.Component<any, any> {
    render() {
        return <div className = "w-screen h-screen bg-no-repeat bg-cover bg-center" style={{
            backgroundImage: `url("https://images.wallpaperscraft.com/image/single/city_skyscrapers_clouds_rain_road_cars_lights_58563_3840x2160.jpg")`
        }}>
            <div className="w-full h-full flex lg:h-screen flex-wrap mx-auto lg:my-0">
                <div className="z-30 absolute right-10 absolute bottom-7">
                    <img className="lg:w-full" src="https://www.artwork.de/wp-content/uploads/2015/08/logo_TF_NEU_4c_ai.png" alt="IHKLogo"/>
                </div>
                <div className="z-20 w-full h-screen lg:w-4/12 mx-0 bg-clip-padding backdrop-filter backdrop-blur-xl bg-opacity-30">
                    <div className="pl-4 pr-4 pb-2 pt-2 grid grid-rows-2 gap-8 mt-2">
                        <div className="w-full h-0 pb-full"/>
                        <SquareHolder/>
                    </div>
                </div>
                <div className="z-10 w-2/3 absolute right-0">
                    <div className="pl-4 pr-4 pb-2 pt-2 grid grid-cols-2 gap-8 mt-2">
                        <SquareHolder />
                        <SquareHolder />
                        <SquareHolder />
                        <SquareHolder />
                    </div>
                </div>
            </div>
        </div>;
    }
}