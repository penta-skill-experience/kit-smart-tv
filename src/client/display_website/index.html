<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Smart TV</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
