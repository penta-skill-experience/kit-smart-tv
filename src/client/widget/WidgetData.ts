export class WidgetData {

    widgetId: string;
    location: number;
    fill: boolean;
    rawConfig: Object;

    constructor(widgetId: string, location: number, fill: boolean, rawConfig: Object) {
        this.widgetId = widgetId;
        this.location = location;
        this.fill = fill;
        this.rawConfig = rawConfig;
    }
}