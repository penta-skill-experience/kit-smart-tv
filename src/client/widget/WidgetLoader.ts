import {Widget} from "./Widget";
import {RssFeedWidget} from "../widget_catalog/RssFeedWidget";

export class WidgetLoader {

    private readonly widgets = new Map<string, Widget>();

    constructor() {
        this.widgets.set("rss-feed", new RssFeedWidget());
    }

    getWidgetIds(): string[] {
        return Array.from(this.widgets.keys());
    }

    getWidget(widgetId: string): Widget {
        if (this.widgets.has(widgetId)) {
            return this.widgets.get(widgetId);
        } else {
            throw new Error(`no widget with ID "${widgetId}" registered`);
        }
    }
}