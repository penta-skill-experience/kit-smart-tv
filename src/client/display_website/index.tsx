import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {RootComponent} from "./RootComponent";

ReactDOM.render(
    <RootComponent/>,
    document.getElementById("root")
);