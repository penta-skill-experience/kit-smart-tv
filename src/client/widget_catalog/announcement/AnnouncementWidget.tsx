import {Widget} from "../../widget/Widget";
import {WidgetConfigSaver} from "../../widget/WidgetConfigSaver";

export class AnnouncementWidget implements Widget {

    createConfigComponent(rawConfig: Object, saver: WidgetConfigSaver): JSX.Element {
        return undefined;
    }

    createDisplayComponent(rawConfig: Object): JSX.Element {
        return undefined;
    }

    getTitle(): string {
        return "Announcements";
    }

    isConfigurable(): boolean {
        return false;
    }

}