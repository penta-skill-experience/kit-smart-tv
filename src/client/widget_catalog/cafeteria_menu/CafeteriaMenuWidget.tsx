import * as React from "react";
import {Widget} from "../../widget/Widget";
import {WidgetConfigSaver} from "../../widget/WidgetConfigSaver";
import {CafeteriaMenu} from "./CafeteriaMenu";

export class CafeteriaMenuWidget implements Widget {

    createConfigComponent(rawConfig: Object, save: (rawConfig: Object) => void): JSX.Element {
        return undefined;
    }

    createDisplayComponent(rawConfig: Object): JSX.Element {
        return <CafeteriaMenu />;
    }

    getTitle(): string {
        return "Cafeteria Widget";
    }

    isConfigurable(): boolean {
        return false;
    }

    getDefaultRawConfig(): Object {
        return undefined;
    }
}