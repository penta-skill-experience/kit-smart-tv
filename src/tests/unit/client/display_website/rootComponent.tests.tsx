import React from "react";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {shallow, configure} from "enzyme";
import toJson from "enzyme-to-json";
import {DesignConfigValues, DesignUtility} from "../../../../shared/persistence/DesignUtility";
import {WidgetPersistence} from "../../../../shared/persistence/WidgetPersistence";
import {WidgetLoader} from "../../../../client/widget/WidgetLoader";
import {RootComponent} from "../../../../client/display_website/RootComponent";
import {ColorScheme} from "../../../../shared/values/ColorScheme";
import {FontSize} from "../../../../shared/values/FontSize";
import {WidgetData} from "../../../../client/widget/WidgetData";
import {RssFeedWidget} from "../../../../client/widget_catalog/rss_feed/RssFeedWidget";
import {CafeteriaMenuWidget} from "../../../../client/widget_catalog/cafeteria_menu/CafeteriaMenuWidget";
import {TramScheduleWidget} from "../../../../client/widget_catalog/tram_schedule/TramScheduleWidget";
import {AnnouncementWidget} from "../../../../client/widget_catalog/announcement/AnnouncementWidget";
import {CafeteriaOpeningWidget} from "../../../../client/widget_catalog/cafeteria_opening/CafeteriaOpeningWidget";

const colorScheme: ColorScheme = {
    id: "dark",
    name: "dark",
    titleFontColor: "black",
    bodyFontColor: "black",
    specialBoldFontColor: "red",
    specialSubtleFontColor: "green",
    accentBarColor: "black",
    backgrounds: ["https://cdn.pixabay.com/photo/2020/04/08/06/55/elephant-5015964_1280.jpg"],
};
const fontSize: FontSize =  {
        id: "medium",
        name: "medium",
        relativeSize: 1,
    };
const values: DesignConfigValues = {
    fontSize: fontSize,
    colorScheme: colorScheme,
    background: colorScheme.backgrounds[0],
};
const widgetOne = new WidgetData("cafeteria-menu", 2, {});
const widgetTwo = new WidgetData("cafeteria-opening", 2, {});
const widgetThree = new WidgetData("announcement", 2, {});
const widgetFour = new WidgetData("tram-schedule", 2, {});
const widgetFive = new WidgetData("cafeteria-menu", 2, {});
const widgetSix = new WidgetData("announcement", 2, {});
const widgetDataList = [widgetOne, widgetTwo, widgetThree, widgetFour, widgetFive, widgetSix];

//mocking display website persistence to return test values for these tests
//no prototype for static methods
const getDesignConfigValuesMock = jest.spyOn(DesignUtility, "getDesignConfigValues");
const getWidgetDataListMock = jest.spyOn(WidgetPersistence.prototype, "getWidgetDataList");
const getWidgetMock = jest.spyOn(WidgetLoader.prototype, "getWidget");

configure({adapter: new Adapter()});

describe("rootComponent Snapshots", () => {

    test("rootComponent Snapshots with design values and widget info", async () => {
        getDesignConfigValuesMock.mockImplementation(() => {
            return new Promise<DesignConfigValues>(resolve => {
                resolve(values);
            });
        });
        getWidgetDataListMock.mockImplementation(() => {
            return new Promise<WidgetData[]>(resolve => {
                resolve(widgetDataList);
            });
        });
        getWidgetMock.mockImplementation((widgetId: string) => {
            let filler = null;
            if(widgetId == "cafeteria-menu") {
                filler = new CafeteriaMenuWidget();
            }
            if(widgetId == "tram-schedule") {
                filler = new TramScheduleWidget();
            }
            if(widgetId == "cafeteria-menu") {
                filler = new CafeteriaMenuWidget();
            }
            if(widgetId == "announcement") {
                filler = new AnnouncementWidget();
            }
            if(widgetId == "cafeteria-opening") {
                filler = new CafeteriaOpeningWidget();
            }
            if(widgetId == "cafeteria-opening") {
                filler = new RssFeedWidget();
            }
            if(filler == null) {
                filler = new RssFeedWidget();
            }
            filler.getTitle();
            filler.isConfigurable();
            filler.getDefaultRawConfig();
            filler.getDefaultRawConfig();
            filler.getConfigComponentClass();
            return filler;
        });
        const wrapper = shallow(<RootComponent/>);
        await new Promise(process.nextTick);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    afterEach(() => {
        jest.restoreAllMocks();
    });
});