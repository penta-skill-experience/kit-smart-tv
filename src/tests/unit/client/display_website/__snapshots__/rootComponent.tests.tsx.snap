// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootComponent Snapshots rootComponent Snapshots with design values and widget info 1`] = `
<div
  className="bg-cover bg-no-repeat bg-center"
  id="mainFrame"
  style={
    Object {
      "backgroundImage": "url(https://cdn.pixabay.com/photo/2020/04/08/06/55/elephant-5015964_1280.jpg)",
      "fontSize": "1rem",
      "height": "100vh",
      "width": "100vw",
    }
  }
>
  <div
    className="flex "
    style={
      Object {
        "color": "black",
      }
    }
  >
    <div
      className="z-50"
      style={
        Object {
          "left": "4.7vh",
          "position": "absolute",
          "top": "45.7vh",
        }
      }
    >
      <img
        alt="IHKLogo"
        className="sm:w-24 lg:w-36 2xl:w-52 4xl:w-72"
        src="https://www.artwork.de/wp-content/uploads/2015/08/logo_TF_NEU_4c_ai.png"
      />
    </div>
    <div
      className="z-20 mx-0 bg-clip-padding backdrop-filter backdrop-blur-xl bg-opacity-30"
      style={
        Object {
          "height": "100vh",
          "width": "33.55vw",
        }
      }
    />
    <div
      className="box-border z-20 grid grid-cols-3 absolute left-0 grid-rows-2 sm:gap-2 md:gap-3 lg:gap-5 xl:gap-7 4xl:gap-10 sm:p-2 md:p-3 lg:p-5 xl:p-7 4xl:p-10"
      style={
        Object {
          "height": "100vh",
          "width": "100vw",
        }
      }
    >
      <div
        className="sm:pl-2 md:pl-3 lg:pl-5 xl:pl-7 4xl:pl-10 grid grid-rows-2 h-full sm:gap-2 md:gap-3 lg:gap-4 xl:gap-6 4xl:gap-8"
      >
        <TimeDisplayComponent
          error={[Function]}
          specialBoldFontColor=""
          specialSubtleFontColor=""
        />
        <WeatherDisplayComponent
          error={[Function]}
          specialBoldFontColor=""
          specialSubtleFontColor=""
        />
      </div>
      <div />
      <RotatorComponent>
        <div
          key="0"
        >
          <SquareHolder
            accentColor="black"
            displayComponentClass={[Function]}
            rawConfig={Object {}}
            specialBoldFontColor="red"
            specialSubtleFontColor="green"
            title="Cafeteria"
            titleColor="black"
          />
        </div>
        <div
          key="1"
        >
          <SquareHolder
            accentColor="black"
            displayComponentClass={[Function]}
            rawConfig={Object {}}
            specialBoldFontColor="red"
            specialSubtleFontColor="green"
            title="Cafeteria Status"
            titleColor="black"
          />
        </div>
        <div
          key="2"
        >
          <SquareHolder
            accentColor="black"
            displayComponentClass={[Function]}
            rawConfig={Object {}}
            specialBoldFontColor="red"
            specialSubtleFontColor="green"
            title="Tram Schedule"
            titleColor="black"
          />
        </div>
        <div
          key="3"
        >
          <SquareHolder
            accentColor="black"
            displayComponentClass={[Function]}
            rawConfig={Object {}}
            specialBoldFontColor="red"
            specialSubtleFontColor="green"
            title="RSS Feed"
            titleColor="black"
          />
        </div>
      </RotatorComponent>
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
